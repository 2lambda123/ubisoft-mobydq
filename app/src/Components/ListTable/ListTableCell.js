import React from 'react';
import TableCell from '@material-ui/core/TableCell';

/**
 * Table cell presentation component
 *
 * @param contents - table cell contents.
 * @param key - Table cell key
 *
 */

export const ListTableCell = ({ contents }) => <TableCell>{contents}</TableCell>;


export default ListTableCell;
