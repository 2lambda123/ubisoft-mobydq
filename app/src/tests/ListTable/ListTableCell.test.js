import React from 'react'
import * as Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ListTableCell from './../../Components/ListTable/ListTableCell'

describe('ListTable unit test', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ListTableCell contents='mock contents'/>)
  });

  it('renders', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ListTable functional test', () => {
  it ('renders correct contents', () => {
    let wrapper = mount(<table><tbody><tr><ListTableCell contents='mock contents'/></tr></tbody></table>);
    expect(wrapper.text()).toEqual('mock contents')
  });
});