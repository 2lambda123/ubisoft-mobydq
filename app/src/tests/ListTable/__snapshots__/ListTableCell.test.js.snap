// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTable unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListTableCell
    contents="mock contents"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "mock contents",
    },
    "ref": null,
    "rendered": "mock contents",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "mock contents",
      },
      "ref": null,
      "rendered": "mock contents",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
